test/integration/sds_dynamic_integration_test.cc

```diff
@@ -40,17 +46,62 @@ namespace Ssl {
 // Hack to force linking of the service: https://github.com/google/protobuf/issues/4221.
 const envoy::service::secret::v3::SdsDummy _sds_dummy;
 
+std::string sdsTestParamsToString(
+    const ::testing::TestParamInfo<std::tuple<Network::Address::IpVersion, Grpc::ClientType, bool>>&
+        p) {
+  return fmt::format(
+      "{}_{}_{}", std::get<0>(p.param) == Network::Address::IpVersion::v4 ? "IPv4" : "IPv6",
+      std::get<1>(p.param) == Grpc::ClientType::GoogleGrpc ? "GoogleGrpc" : "EnvoyGrpc",
+      std::get<2>(p.param) ? "UsesQuic" : "UsesTcp");
+}
+
+std::vector<std::tuple<Network::Address::IpVersion, Grpc::ClientType, bool>>
+getSdsTestsParams(bool test_quic) {
+  std::vector<std::tuple<Network::Address::IpVersion, Grpc::ClientType, bool>> ret;
+  for (auto ip_version : TestEnvironment::getIpVersionsForTest()) {
+    for (auto grpc_type : {Grpc::ClientType::EnvoyGrpc, Grpc::ClientType::GoogleGrpc}) {
+#ifndef ENVOY_GOOGLE_GRPC
+      if (grpc_type == Grpc::ClientType::GoogleGrpc) {
+        continue;
+      }
+#endif
+
+      ret.push_back({ip_version, grpc_type, false});
+      if (test_quic) {
+#ifdef ENVOY_ENABLE_QUIC
```

Comment:
I also think we should have one for client / upstream protocols but I can take that on, since I didn't add the correct utility over in https://github.com/envoyproxy/envoy/pull/15853/files